### PR TITLE
app-arch/lz4: version bump to 0_p131

### PR DESCRIPTION
**Pull request by eliasp**
Package-Manager: portage-2.2.23

As a prerequisite for an upcoming PR to make `sys-apps/systemd-9999[lz4]` build again, a newer `app-arch/lz4` version is required.

The ebuild is a copy of `lz4-0_p120.ebuild`, but with `KEYWORDS` reset to `~` of the only arch I could test it on (`~amd64`).

### CODE REVIEW

**Comment by hasufell** [comment]
@ryao

**Comment by floppym** [comment]
Please double check this for ABI compatibility; I seem to recall that they changed/dropped some symbols. If so, a subslot should be added to the ebuild.

**Comment by floppym** [comment]
Here's the detail on that ABI breakage.

https://code.google.com/p/lz4/issues/detail?id=147

**Comment by floppym** [comment]
I would suggest setting `SLOT="0/${PV}"` since the upstream developer does not understand the significance of maintaining ABI compatibility.

**Comment by eliasp** [comment]
Updated the PR as suggested.

**Comment by mgorny** [comment]
@floppym AFAICS upstream said there will be no major ABI changes now. Maybe we should stick with specific subslot instead? Or try to pull request upstream into bumping SONAME.

**Comment by floppym** [comment]
@mgorny I'm not sure how you came to that conclusion.

I see no mention of ABI stability; rather, I see a long-winded explanation of some header file changes.

I also see a refusal to make the SONAME anything other than the library API major version.

**Comment by mgorny** [comment]
Yeah. Should we force-change SONAME to something Gentoo-specific then? like "g${PV}" or "g$(next-number)"?

**Comment by floppym** [comment]
> Should we force-change SONAME to something Gentoo-specific then? like "g${PV}" or "g$(next-number)"?

So long as we update all revdeps with slot-operators, it won't really be an issue for ebuilds.

I suppose it might be useful for people building stuff against lz4 outside of portage.

If we go that route, we should probably do some real analysis on how the ABI has changed (if at all). Maybe I'm overreacting.

http://lvc.github.io/abi-compliance-checker/

**Comment by ryao** [comment]
@eliasp Thanks for this pull request. I am travelling and I have some other things that I need to do before I get on my plane tomorrow. I can try to review it sooner if necessary, but I would rather wait until after my plane flight.

**Comment by mgorny** [comment]
@ryao, don't worry, me and @floppym can handle it for you once we figure out all ABI stuffs to worry about ;-).

**Comment by floppym** [comment]
I'm in no great rush to get this merged; it can certainly wait another day or two.

**Comment by hasufell** [comment]
I think `${PV}` in subslot should really be avoided. If upstream doesn't check for ABI compatibility, then we have to instead of assuming that it always changes.

**Comment by mgorny** [comment]
Ok, I guess I should bump acc.

File: app-arch/lz4/lz4-0_p131.ebuild [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+		sed -i -e '/^test:/s|test-mem||g' programs/Makefile || die
+	fi
+	multilib_copy_sources
+}
+
+multilib_src_compile() {
+	tc-export CC AR
+	# we must not use the 'all' target since it builds test programs
+	# & extra -m32 executables
+	emake
```
Comment (mgorny) [inline app-arch/lz4/lz4-0_p131.ebuild]: Ok, so first nitpick: I'm pretty sure we don't want to build it all on alternate ABIs. I guess some combination of `multilib_is_native_abi` and `use test` would be desired.

**Comment by mgorny** [comment]
Did an ACC run on 120 vs 131, and they're binary-incompatible. So subslot and SONAME bump is desired.

File: app-arch/lz4/lz4-0_p131.ebuild (hunk 1 above)
Comment (eliasp) [inline app-arch/lz4/lz4-0_p131.ebuild]: Could one of you devs take over at this point? Missing the required knowledge/background here…

**Comment by ryao** [comment]
@mgorny @floppym I was suffering from jetlag after returning to New York and had other things backlogged. I intend to get to this and I am now shooting for the end of the week, but I do not want to impose an arbitrarily long delay in the event that I cannot keep that target. If I do not handle this by midnight Sunday morning EST, feel free to call maintainer timeout and commit, provided that you both agree on the changes. I have confidence in both of you and if a mistake gets by at least two of us, we can deal with it afterward.

**Comment by mgorny** [comment]
@floppym, so what do we do here? Custom SONAME?

**Comment by floppym** [comment]
Yeah, let's do that.

File: app-arch/lz4/lz4-0_p131.ebuild (hunk 1 above)
Comment (mgorny) [inline app-arch/lz4/lz4-0_p131.ebuild]: Well, it's not really that important, so we'll pass until someone decides to improve it ;-).

**Comment by mgorny** [comment]
After some testing, I'm no longer convinced this is the best thing to do. If we change SONAME, all binary packages linking against liblz4 will be of no use to Gentoo. Unless we do compatibility symlinks ­— but then, why bump SONAME in the first place? Maybe we should just bump `SLOT` every time ABI changes and live with the potential SONAME conflicts.

**Comment by floppym** [comment]
I don't feel strongly either way; there are downsides to changing or not changing soname.

I would lean toward not changing soname at the distro level for the sake of simplicity.

**Comment by mgorny** [comment]
I'm sorry but I pretty much ended up doing this afresh. Makefile use needed more fixes, keywords were dropped unnecessarily and I've set the subslot to r131 with an appropriate note. I've also updated the live ebuild.

**Pull request closed without merging**